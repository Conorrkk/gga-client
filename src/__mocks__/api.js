const teamMock = {
    teamName: 'Bredagh',
    teamLevel: 'Senior',
}

export const apiRequests = () => {
    return {
        getTotalGoals: jest.fn(() => 2),
        getTotalPoints: jest.fn(() => 10),
        getTeamById: jest.fn(() => {
            return teamMock;
        })
    };
}