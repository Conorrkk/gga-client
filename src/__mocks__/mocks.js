export const player1 = {
    _id: 1,
    playerId: 1,
    playerName: 'Conor Kelly',
    stats: {
        goal_from_play: 0,
        goal_from_dead: 1,
        point_from_play: 2,
        point_from_dead: 0,
        wide: 0
    }
}

export const player2 = {
    _id: 2,
    playerId: 2,
    playerName: 'Harry Potter',
    stats: {
        goal_from_play: 0,
        goal_from_dead: 1,
        point_from_play: 2,
        point_from_dead: 0,
        wide: 2
    }
}

export const playersMock = [player1, player2]; 

export const match = {
    _id: "123",
    teams: {
        teamId: "123",
        oppositionTeam: "Galway",
        players: [player1, player2]
    },
    matchDate: 8/30/2023,
    goalAgainst: 1,
    pointAgainst: 14
};

export const matches = [match];