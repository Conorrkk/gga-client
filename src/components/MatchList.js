import MatchShow from "./MatchShow";
import { Container } from "react-bootstrap";

function MatchList({ matches, onDelete }) {
  const loadedMatches = matches.map((match) => {
    return <MatchShow onDelete={onDelete} key={match._id} match={match} />;
  });

  return <Container fluid >{loadedMatches}</Container>;
}

export default MatchList;
