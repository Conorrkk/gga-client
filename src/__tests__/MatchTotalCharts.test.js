import React from 'react';
import renderer from 'react-test-renderer';
import MatchList from '../components/MatchList';
import { matches } from '../__mocks__/mocks';
import { BrowserRouter } from 'react-router-dom';
import MatchTotals from '../components/MatchTotalsChart';

const dataMock = [0.5, 0.7, 1];

it('renders correctly', () => {
    const tree = renderer
    .create(<MatchTotals data={dataMock}/>)
    .toJSON();

    expect(tree).toMatchSnapshot();
});