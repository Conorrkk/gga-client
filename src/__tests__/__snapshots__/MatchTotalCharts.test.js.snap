// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <canvas
    height={150}
    role="img"
    style={
      Object {
        "margin": "10px",
        "width": "90%",
      }
    }
    width={300}
  />
</div>
`;
